bzt/engine.py

```diff
@@ -1059,3 +1059,62 @@ def __init__(self, config, engine):
                     body = fhd.read()
 
         self.body = body
+
+
+class PythonGenerator(object):
+    IMPORTS = ''
+
+    def __init__(self, scenario, parent_logger):
+        self.root = etree.Element("PythonCode")
+        self.tree = etree.ElementTree(self.root)
+        self.log = parent_logger.getChild(self.__class__.__name__)
+        self.scenario = scenario
+
+    def add_imports(self):
+        imports = etree.Element("imports")
+        imports.text = self.IMPORTS
+        return imports
+
+    @abstractmethod
+    def build_source_code(self):
+        pass
+
+    @staticmethod
+    def gen_class_definition(class_name, inherits_from, indent="0"):
```

Comment:
Having `indent` to be a string with numeric value inside is pretty confusing. Can we use ints on the interface, while passing strings to lxml?